src/transformers/generation_tf_utils.py

```diff
@@ -756,6 +759,8 @@ def generate(
             )
             # expand encoder_outputs
             encoder_outputs = (tf.gather(encoder_outputs[0], expanded_batch_idxs, axis=0),)
+            if attention_mask is not None:  # vision models don't have this
+                attention_mask = tf.gather(attention_mask, expanded_batch_idxs, axis=0)
```

Comment:
Is this only needed for the new model or was it a bug to not have done this previously?

Reply:
I'm a bit worried about this as it could lead to some unexpected errors in the more complicated models like TFRag. Could we maybe run the following tests here to double check that this change is ok:

`RUN_SLOW=1 pytest tests/test_modeling_tf_bart.py`
`RUN_SLOW=1 pytest tests/test_modeling_tf_t5.py`
`RUN_SLOW=1 pytest tests/test_modeling_tf_rag.py` (not that you need some more dependencies to be installed here)

Reply:
Spot on, it failed for those models. 

I've reworked these lines to behave like pytorch's `_expand_inputs_for_generation()`, which is used in all modalities. I've tested the changes against bart, bert, t5, vit, and s2t -- let me know what you think, @patrickvonplaten

Reply:
Cool! Could you please also run the slow `TFRag` tests - it's a bit annoying as it's a huge and slow test, but it's the model that is most prone to break if there are changes in tf generate

Reply:
Can confirm it passes TFRag tests as well (had to spin up a larger machine 😅 )

Reply:
Cool! Good for me then